src/config.rs

```diff
@@ -189,6 +259,17 @@ mod test {
                     .unwrap();
             assert_eq!(gitlab, new_gitlab("user/repo", version("0.1.0")));
         }
+
+        #[test]
+        fn duplicate_tools() {
+            let err = toml::from_str::<ConfigFileTools>(
+                r#"tool = { github = "user/repo", version = "0.1.0" }
+			tool = { github = "user2/repo2", version = "0.2.0" }"#,
+            )
+            .unwrap_err();
+
+            assert_eq!(err.to_string(), "duplicate tool `tool` at line 1 column 1");
+        }
```

Comment:
Added these test cases now.

```bash
❯ cat foreman.toml


[tools]
# oops!
selene = { source = "rojo-rbx/rojo", version = "=7.1.1" }

selene = { source = "Kampfkarren/selene", version = "=0.19.1" }
❯ selene --version
unable to parse Foreman configuration file (at ~/project/foreman.toml): duplicate tool name `selene` found for key `tools` at line 3 column 1

A Foreman configuration file looks like this:

[tools] # list the tools you want to install under this header

# each tool is on its own line, the tool name is on the left
# side of `=` and the right side tells Foreman where to find
# it and which version to download
tool_name = { github = "user/repository-name", version = "1.0.0" }

# tools hosted on gitlab follows the same structure, except
# `github` is replaced with `gitlab`

# Examples:
stylua = { github = "JohnnyMorganz/StyLua", version = "0.11.3" }
darklua = { gitlab = "seaofvoices/darklua", version = "0.7.0" }
```

So I think it's not really an issue, it's just that the line serde points to is where `[tools]` is in the toml, rather than where the duplicate tool is. Unsure if that is a big enough issue to warrant looking in to.